src/celeritas/optical/OpticalStepCollector.hh

```diff
@@ -0,0 +1,103 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/OpticalStepCollector.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "celeritas/phys/ParticleParams.hh"
+#include "celeritas/user/StepInterface.hh"
+
+#include "OpticalDistributionData.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Callback class derived from `StepInterface` that stores
+ * \c OpticalStepCollectorData to be used by optical physics pre-generator
+ * classes.
+ *
+ */
+class OpticalStepCollector final : public StepInterface
```

Comment:
Didn't we decide to postpone the implementation of this class (since we might end up with something that's like the step collector but different)? This class isn't used or tested anywhere in the PR...

Reply:
Sorry, maybe I misunderstood the conclusion of our chat. We discussed that the existing `StepInterface` is not ideal for the optical physics, but we wouldn't make a new one as of now so I kept this in to have a starting point. I can remove it from this PR if you prefer.

Reply:
I could've misunderstood or forgotten myself :)

Since it's not necessary to set up the step data in the tests, and it's not tested or used anywhere, let's just remove it for now. We can `git revert` the removal in a follow-on PR where we actually test the integration into the tracking loop.

Reply:
Great. Sounds good to me.